src/games/takesix/game.ts

```diff
@@ -0,0 +1,219 @@
+      return INVALID_MOVE;
+    }
+
+    // If card is #6 move all cards from deck to player's hand
+    if (G.decks[id].length === 5) {
+      return {
+        ...G,
+        players: Object.values({
+          ...G.players,
+          [ctx.playerID]: {
+            ...G.players[ctx.playerID as any],
+            penaltyCards: [
+              ...G.players[ctx.playerID as any].penaltyCards,
+              ...G.decks[id],
+            ],
+          },
+        }),
+        decks: Object.values({
+          ...G.decks,
+          [id]: [card],
+        }),
+      };
+    }
+
+    return {
+      ...G,
+      decks: Object.values({
+        ...G.decks,
+        [id]: [...G.decks[id], card],
+      }),
+    };
+  }
+}
+
+export const TakeSixGame = Game({
+  name: 'takesix',
+  flow: {
+    endTurn: false,
+    endPhase: false,
+    endGame: false,
+    startingPhase: 'CARD_SELECT', // Start by selecting cards
+    phases: {
+      // Everyone needs to select card
+      CARD_SELECT: {
+        allowedMoves: ['selectCard'],
+        turnOrder: TurnOrder.ANY_ONCE,
+        next: 'DECK_SELECT',
+        // Determine player order
+        onPhaseEnd: (G: IG) => {
+          const selectedCards = G.players.map(
+            (player: Player) => player.selectedCard,
+          );
+          selectedCards.sort((a: Card, b: Card) => a.number - b.number);
+          return {
+            ...G,
+            cardOrder: selectedCards
+              .map((card: Card) => card.owner)
+              .map((x: number) => x.toString()),
+          };
+        },
+      },
+      // Select deck
+      DECK_SELECT: {
+        allowedMoves: ['selectDeck'],
+        next: 'CARD_SELECT',
+        // Implement CUSTOM_FROM_ONCE
+        turnOrder: {
+          playOrder: (G: IG) => G.cardOrder,
+          first: () => 0,
+          next: (_, ctx) => {
+            if (ctx.playOrderPos < ctx.playOrder.length - 1) {
+              return ctx.playOrderPos + 1;
+            }
+          },
+        },
+        onMove: (_, ctx) => {
+          ctx.events.endTurn();
+        },
+        onPhaseEnd: (G: IG) => {
+          if (G.players[0].cards.length === 0) {
+            G.end = true;
+          }
+        },
+      },
+    },
+    endGameIf: (G: IG) => {
+      if (G.end === true) {
+        const scoreboard = G.players.map((player, i) => ({
+          playerID: i,
+          penaltyPoints: player.penaltyCards.reduce(
+            (acc, card) => acc + card.value,
+            0,
+          ),
+        })).sort((a, b) => a.penaltyPoints - b.penaltyPoints);
+
+        if (scoreboard[0].penaltyPoints === scoreboard[1].penaltyPoints) {
+          return { draw: true };
+        } else {
+          return { winner: scoreboard[0].playerID.toString() };
+        }
+      }
+    },
+  },
+  // playerView: PlayerView.STRIP_SECRETS,
+  setup: (ctx): IG => {
+    // Generate deck
+    const deck = ctx.random.Shuffle(
+      new Array(104).fill(0).map((_, i) => {
```

Comment:
wouldnt it be clearer if you do:

[ 
  ...new Array(20).fill(2),
  ...new Array(10).fill(3)

and etc?

Reply:
Can you explain it little bit more? I don't get it 😄

Reply:
Your function generates 76 cards with value 1, 9 cards with value 2, 10 cards with value 3,  8 cards with value 5, and 1 card with value 7. Then, you shuffle the cards. It would be easier to understand if this was generated more directly instead of using the % function. The code would look like this:
```
const deck = ctx.random.Shuffle([ 
    ...new Array(76).fill(1), 
    ...new Array(9).fill(2), 
    ...new  Array(10).fill(3), 
    ...new  Array(8).fill(5), 
    7 
].map((value, i) => new Card(i+1, value, null));
```

Reply:
Oh, I just noticed that my code is not the same as yours, it would have the wrong card "number". Interesting.

Reply:
I just noticed that in the wikipedia theses cards are defined with the multiples. So your implementation makes sense. Sorry for the noise!